### ISSUE TITLE
Retest torch 2.11+ and lift the <2.11 cap

### ISSUE THREAD

**Issue by jgamblin** [RogoLabs/CVEForecast#29]
Stage 2 of the dependency modernization (#27) added a \`torch>=2.10,<2.11\` cap in \`requirements.txt\` because torch 2.11.0 combined with pandas 3.0.2 caused a segfault during XGBoost training in the production forecast pipeline.

## When to retest

- After the next torch patch release (2.11.1+) or torch 2.12, **or**
- After the next pandas patch (3.0.3+), **or**
- ~3 months from now (2026-07-28) — whichever comes first.

## Retest procedure

1. \`git checkout -b deps/lift-torch-cap\`
2. Edit \`requirements.txt\`: change \`torch>=2.10,<2.11\` to \`torch>=2.11\` (or latest)
3. \`pip install --upgrade -r requirements.txt\`
4. \`python code/run_production_forecast.py\` — must complete without segfault
5. \`pytest -v\` — must stay 27/27
6. If green, PR and merge; if segfault still occurs, check torch + pandas versions in play and update this issue with the new combo that reproduces.

## Context

- Spec: \`docs/superpowers/specs/2026-04-28-dependency-modernization-design.md\`
- Plan: \`docs/superpowers/plans/2026-04-28-dependency-modernization.md\`
- Stage 2 PR: #27

**Comment by jgamblin** [comment]
Retest complete — **cap lifted to `torch>=2.14,<2.15`** in #51.

## Step 4 result

The procedure's step 4 (production pipeline, no segfault) could not be run locally: this machine is on Python 3.14, which has no torch wheels. It ran on CI instead, [run 35261256549](https://github.com/RogoLabs/CVEForecast/actions/runs/35261256549) on Python 3.13:

```
Downloading torch-2.14.0+cpu-cp313-cp313-manylinux_2_28_x86_64.whl (196.3 MB)
...
PRODUCTION FORECAST COMPLETE
🎉 All forecasts generated successfully!
```

No segfault. Conclusion: success.

## Versions in play

| | capped at | now |
|---|---|---|
| torch | 2.10.x | **2.14.0+cpu** |
| pandas | 3.0.2 (when it segfaulted) | 3.0.3 |
| darts | | 0.47.0 |
| Python | | 3.13 (CI) |

## Notes for next time

Dependabot's own attempt (#49) could not install at all — it generated `torch>=2.14.0+cpu`, and pip rejects a local version label with `>=`. The `--extra-index-url` at the top of `requirements.txt` already resolves the CPU wheels, so no suffix is needed; `torch>=2.14,<2.15` is the correct form.

Worth recording that "torch models are disabled in config.json" is **not** a reason to consider a torch bump safe: darts imports torch whenever it is installed, and the original segfault was during XGBoost training with torch merely present. Having it installed is the risk factor, so any future bump needs this same pipeline run rather than a green test job.
